### PR TITLE
Improve Docker/Dev Experience closes #19

### LINKED ISSUE #19: Improve Development Experience with Docker
Add Docker containers and scripts for development, CI, and Test environments.

-----
**TODO**
- [x] Create Dockerfile (see below)
- [x] Remove `bundle install` from `package.json` `postinstall` or make it optional for compatibility with Docker build container
- [ ] ~~Create Docker Container hosted on public Docker Hub using Dockerfile (one less step for devs)~~ Removed until investigation of shared RAX account.
- [x] Fix `Gemfile.lock` or remove it
- [x] Update docs

-----
**Running Docs Locally**
**Note:** Remove current `Gemfile.lock`
```
docker run --rm --label=jekyll --volume=$(pwd):/srv/jekyll -it -p 9000:4000 jekyll/jekyll jekyll serve --watch
```
-----
**Grunt/Bower Build Dockerfile** - Adapted from Jekyll Dockerfile + Grunt & Bower
**Pending:** `npm install` at run time

```
FROM envygeeks/alpine
MAINTAINER "Brady Vitrano <brady.vitrano@rackspace.com>"
RUN apk --update add build-base \
                     ruby ruby-dev ruby-io-console ruby-irb ruby-json ruby-rake ruby-rdoc \
                     nodejs \
                     && \

    mkdir -p /home/bob && \
    addgroup -Sg 1000 bob &&  \
    adduser  -SG bob -u 1000 -h /home/bob bob && \
    chown bob:bob /home/bob && \

    cd ~ && \
    yes | gem update --system --no-document -- --use-system-libraries && \
    yes | gem update --no-document -- --use-system-libraries && \

    cd ~ && \
    gem install sass --no-document && \

    npm install -g bower grunt-cli && \
    echo '{ "allow_root": true }' > /root/.bowerrc && \
    gem clean && \
    apk del build-base ruby-rdoc ruby-dev && \

    mkdir -p /src/app && \
    chown bob:bob /src/app && \
    echo 'bob ALL=NOPASSWD:ALL' >> /etc/sudoers && \
    rm -rf /usr/lib/ruby/gems/*/cache/*.gem && \
    rm -rf /var/cache/apk/* && \
    docker-helper cleanup

WORKDIR /src/app
```

### PR DESCRIPTION
**Pull request by brady-vitrano**
Closes  #19. This PR assumes that a developer only has Docker installed. NPM, Grunt, Jekyll, and all other tasks are run within containers. Detailed workflow is defined in the README.md. 

Here are the changes that I made:

* Added `dist` folder to version control
* Added `bootstrap` and `jquery` to NPM for faster `webpack` build times
* Created `dockerfiles` directory for relevant Dockerfiles and scripts
* Removed `bundle install` from `package.json`
* Explicit running of NPM and Bower install at container run time
* Created `docker-compose.yml` for managing the containers

Future improvements:

* Install NPM and Bower into the container for possible speed improvements

### CODE REVIEW

File: dockerfiles/Grunt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    adduser  -SG bob -u 1000 -h /home/bob bob && \
+    chown bob:bob /home/bob && \
+
+    cd ~ && \
+    yes | gem update --system --no-document -- --use-system-libraries && \
+    yes | gem update --no-document -- --use-system-libraries && \
+
+    cd /home/bob && \
+    gem install sass --no-document && \
+
```
Comment (brady-vitrano) [inline dockerfiles/Grunt]: Add the following for firewall/proxy connections 

`git config --global url."https://".insteadOf git://`

File: README.md [hunk 2]
```diff
@@ -5,40 +5,85 @@ Derek Zoolander's Corporate Style Guide to Help Product Teams Build Ridiculously
...
-grunt build:derek
+## Restarting Development
+
+```
+cd zoolander
+docker-machine start default
+docker-machine ls
+eval "$(docker-machine env default)"
+docker-compose up -d
+docker-compose ls
```
Comment (brady-vitrano) [inline README.md]: `docker-compose ps`

File: README.md [hunk 3]
```diff
@@ -5,40 +5,85 @@ Derek Zoolander's Corporate Style Guide to Help Product Teams Build Ridiculously
...
 ```
-gem install bundler jekyll
-npm install
+git clone https://github.com/rackerlabs/zoolander.git
+cd zoolander
+docker-machine create --driver=virtualbox --virtualbox-cpu-count=2 --virtualbox-memory=2048 default
+docker-machine ls # get ip address
+eval "$(docker-machine env default)"
+docker-compose up -d
+docker-compose ls # get generated port for 4000
```
Comment (brady-vitrano) [inline README.md]: `docker-compose ps`

**Comment by steveortiz** [comment]
Looks great @brady-vitrano!  Merging.

**Pull request merged**